File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/company/dtos/IntervalSummary.java
package com.slotify.backend.spring.company.dtos;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@ToString
public class IntervalSummary {

    private UUID intervalId;
    private LocalDateTime startDateTime;
    private LocalDateTime endDateTime;
    private LocalDateTime createdAt;
    private Integer maxConcurrentServices;
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/company/IntervalSummary.kt
package com.raj.slotify.dtos.company

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime
import java.util.UUID

data class IntervalSummary(

    @SerializedName("intervalId")
    val intervalId: UUID,

    @SerializedName("startDateTime")
    val startDateTime: LocalDateTime,

    @SerializedName("endDateTime")
    val endDateTime: LocalDateTime,

    @SerializedName("createdAt")
    val createdAt: LocalDateTime,

    @SerializedName("maxConcurrentServices")
    val maxConcurrentServices: Int

)
